### PR TITLE
6937: Replace Graphviz with Lineage Graph Component

### PR DESCRIPTION
**Pull request by labkey-nicka**
Closing and re-issuing PR. See #1031.

### CODE REVIEW

File: experiment/package.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
     }
   },
   "dependencies": {
-    "@labkey/components": "0.41.0"
+    "@labkey/components": "0.42.0-fb-rungraph-lineage.4"
```
Comment (cnathe) [inline experiment/package.json]: I haven't looked at your changes in the labkey-ui-component side for this package version yet, but do you think there is any need to bump the @labkey/components version in the other platform package.json files to keep them in sync? or is this set of change very specific to only the usage in experiment?

File: experiment/src/client/RunGraph/util.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
+export interface AppContext {
+    lsid: string
+    rowId: number
+    target: string
+}
+
+export interface InitAppEventDetail<T> {
```
Comment (cnathe) [inline experiment/src/client/RunGraph/util.ts:7]: The plan is to fine somewhere more central to share the code below this line related to the registerApp stuff, right? Have you thought more about where that might go?

**Review by cnathe — COMMENTED** [review]
The platform webpack config changes look good and the RunGraph React implementation looks good with the registerApp option (I've added this as a Frontend Dev meeting topic to review / discuss / make the group aware). 

I'll let Kevin do the final review / approval of the Java and jsp changes.

File: experiment/src/client/RunGraph/util.ts (hunk 2 above)
Comment (labkey-nicka) [reply experiment/src/client/RunGraph/util.ts:7]: We don't really have a good home for it at the moment. I think `@labkey/api` could suffice as long as the implementation remains agnostic of React, etc. `@labkey/components` could work but it feels more odd as it is not a component and more of a shim/util to help load apps in LKS.

If we port the implementation as-is we'll need a way to ensure the page registers the event listener before pages request to load the app.

File: experiment/package.json (hunk 1 above)
Comment (labkey-nicka) [reply experiment/package.json]: Good idea. I've gone ahead and bumped all usages of `@labkey/components` in platform.

File: .gitattributes [hunk 3]
```diff
@@ -2298,6 +2298,7 @@ experiment/src/org/labkey/experiment/controllers/exp/ExperimentRunGraphView.java
 experiment/src/org/labkey/experiment/controllers/exp/ExperimentRunGroupsView.java -text
 experiment/src/org/labkey/experiment/controllers/exp/ExperimentUpdateView.java -text
 experiment/src/org/labkey/experiment/controllers/exp/exportSampleSetAsXar.jsp -text
+experiment/src/org/labkey/experiment/controllers/exp/experimentRunGraphView.jsp -text
```
Comment (labkey-kevink) [inline .gitattributes]: I don't think we need to add new files to the .gitattributes list -- only for the legacy files moved from svn.  @labkey-klum or @labkey-tchad may know.

**Review by labkey-kevink — APPROVED** [review]

File: .gitattributes (hunk 3 above)
Comment (labkey-adam) [reply .gitattributes]: That's my understanding as well

Comment (labkey-nicka) [reply .gitattributes]: Fixed. Thanks for the info.

**Pull request closed without merging**